src/views/docs-view/components/docs-page-heading/docs-page-heading.module.css

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+.root {
+	align-items: flex-start;
+	display: flex;
+
+	/* DOM order is version selector, then title. On small viewports, with
+	   wrapping, we have multiple rows so row-reverse has no effect.
+		 On large viewports, where we have space to put everything in one row,
+		 we show the version selector after the landingHero (if used). */
+	flex-direction: row-reverse;
```

Comment:
❓ Could be an issue here with visual order not reflecting DOM order? One option we did to work around this in the footer (with the give feedback button) was render two different components and `display:none` for one based on viewport so that the markup matched the visual order.

Reply:
Great callout! 💯 

This may or may not be an depending on our interpretation of the "meaningful sequence" success criterion and related techniques. There's a bit more detail in [this GitHub comment](https://github.com/hashicorp/dev-portal/pull/1687#discussion_r1127047577). I'll go into even more detail below in case it's helpful.

The TLDR is:
- **I'd prefer to handle the DOM-vs-visual-order as a separate ticket**, as this PR maintains the upstream approach to DOM vs visual ordering for the version select.
- The **WCAG pages themselves also "break"** under what I understand to be our very strict interpretation of the DOM-must-match-visual-order technique, so **I think our current interpretation warrants some discussion**.

---

## Meaningful sequence success criteria have some subtlety in technique

We _could_ take a really direct and strict interpretation of the [Making the DOM order match the visual order](https://www.w3.org/WAI/WCAG21/Techniques/css/C27) technique. In other words, we could be very strict that  "anything visually to the left must be earlier the DOM order, anything visually to the right must be later in the DOM order".

But, I think this is not necessary to meet the success criterion we've committed to, and may in fact compromise the user experience for many users, especially those who use keyboard navigation. [Making the DOM order match the visual order](https://www.w3.org/WAI/WCAG21/Techniques/css/C27) is _one_ technique to meet the [Meaningful Sequence](https://www.w3.org/WAI/WCAG21/Understanding/meaningful-sequence) criterion, but it's not the _only_ technique. As noted on the criterion page:

> Each numbered item in this section represents a technique or combination of techniques that the WCAG Working Group deems sufficient for meeting this Success Criterion. However, **it is not necessary to use these particular techniques**.

As an abstract example from [the WCAG success criterion page on meaningful sequences](https://www.w3.org/WAI/WCAG21/Understanding/meaningful-sequence#examples), it can be okay for the visual presentation to not match the programmatically determined order if it doesn't affect the meaning of the page:

> Example 2: CSS is used to position a navigation bar, the main story on a page, and a side story. The **visual presentation of the sections does not match the programmatically determined order, but the meaning of the page does not depend on the order of the sections**. 

To me, this seems logical. For example, whether the table-of-contents or the page content appears first in "visual order" does not affect the meaning of the page, so we can _choose_ to match that visual order to the DOM order (or not!) based on what we think will produce the ideal experience.

As a concrete example of exactly this case, the WCAG page itself places its table-of-contents-style navigational sidecar visually _to the right_ of the main content area (arguably _later_ in the visual order), while that `aside` element appears _earlier_ in the DOM order:

![CleanShot 2023-04-17 at 19 51 44@2x](https://user-images.githubusercontent.com/4624598/232635375-57136bcc-9b8d-4797-91fd-45c90cff1075.png)

I believe this is objectively okay with relation to the "Meaningful sequence" success criteria because the meaning of the page does not depend on the order of the `aside` and `main` sections. And it seems subjectively preferable because putting the `aside` _before_ the `main` element in the DOM order arguably makes for a better user experience, as we can keyboard-navigate through the table-of-contents _before_ having to keyboard-navigate through the entire main content. Quick screen-share of tabbing through things to demonstrate:

https://user-images.githubusercontent.com/4624598/232635346-a7509e86-a231-44d1-b068-7eb581f0b430.mp4

My thinking is that we're in a similar situation with the version select and the `h1`. The visual order of the version selector and the page content does not impact meaning, so I don't think we're strictly required to match it to DOM order. And I think we should in fact _avoid_ strictly matching them, as varying the DOM order seems like it could _negatively_ impact the user experience on the page. Encountering the `h1`, _then_ the version select, _then_ the text content meant to directly follow the `h1` seems non-intuitive to me - especially since it would happen on some viewports but not others (more on this below!).

## Viewport-toggled elements seem non-ideal

All that being said, I can _maybe_ imagine a case for wanting to be very strict about visual order matching DOM order, just as some kind of safeguard.

If this is the case, and we establish team consensus to be strict on that technique front rather than on the success criterion front, **I think we need to solve DOM and visual ordering mismatch problems at the design level**, not the implementation level.

I think by toggling elements on "mobile" vs "desktop" in order to match a strict, meaning-impact-agnostic interpretation of the "Meaningful Sequences" success criterion, we risk introducing additional implementation and end-user complexity by effectively varying the DOM order between "mobile" and "desktop", and have elements pop in and out of existence at different viewport sizes.

As an example of why I think toggling "mobile" and "desktop" elements could be non-ideal, consider the WCAG example as context:

> If a blind user, who reads the page with a screen reader that follows the source order, is working with a sighted user who reads the page in visual order, they may be confused when they encounter information in different orders.

In an analogous example, imagine one user is using working with a slightly smaller viewport and another user is working with a slightly larger viewport (or even, imagine a single user has just resized their viewport slightly, and a second ago had been tabbing through the page). For each viewport in isolation, the visual and DOM order may be more "perfectly matched", but the difference in DOM and focus order across arbitrary viewport breakpoints seems likely to cause some confusion as well. The order is arguably a _more meaningful sequence_ if it is semantically consistent at the programmatic document level across viewports.

So I think ideally, assuming we first establish that we need to strictly match DOM and visual order regardless of impact on page meaning, we'd back this up and come up with a design solution that _avoids_ the order "mismatch" issue altogether, rather than trying to "correct" it in implementation.